lib/puppet/provider/azure_resource_template/arm.rb

```diff
@@ -12,7 +12,7 @@
   def self.instances # rubocop:disable Metrics/AbcSize
     begin
       PuppetX::Puppetlabs::Azure::ProviderArm.new.get_all_deployments.collect do |dep|
-        params = dep.properties.parameters.each_with_object({}) do |(k,v),memo|
+        params = (dep.properties.parameters || {}).each_with_object({}) do |(k,v),memo|
```

Comment:
this is a pretty convoluted way to say that `params` should be empty if `dep.properties.parameters` is nil.

Can you do something like this instead?

```ruby
iterator = dep.properties.parameters || {}
params = iterator.each_with_object({}) do |(k,v),memo|
...
```